### PR TITLE
Moved enrollmentId method to MSIDAuthority

### PR DESCRIPTION
**Pull request by rokhrest**
Moved enrollmentId method to MSIDAuthority.

### CODE REVIEW

File: IdentityCore/src/validation/MSIDAuthority.m [hunk 1]
```diff
@@ -149,6 +150,16 @@ - (NSURL *)universalAuthorityURL
     return @[self.environment];
 }
 
+- (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDAuthority.m]: Shouldn't this be present in AAD authority only?

File: IdentityCore/src/validation/MSIDAuthority.m [hunk 2]
```diff
@@ -149,6 +150,16 @@ - (NSURL *)universalAuthorityURL
...
+{
+    if (homeAccountId != nil)
+    {
+        return [[MSIDIntuneEnrollmentIdsCache sharedCache] enrollmentIdForHomeAccountId:homeAccountId context:nil error:nil];
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDAuthority.m]: Please also check with legacy userId, if none found with home account id.

File: IdentityCore/src/validation/MSIDAuthority.m (hunk 1 above)
Comment (rokhrest) [reply IdentityCore/src/validation/MSIDAuthority.m]: By the design doc and the ADAL code snippets, it looks like we only exclude ADFS authorities. If this behavior is not correct, I will be happy to fix :)

Comment (oldalton) [reply IdentityCore/src/validation/MSIDAuthority.m]: ADAL only supports ADFS and AAD authorities, whereas MSAL also supports B2C :) Enrollment ID will be only applicable to AAD.

Comment (rokhrest) [reply IdentityCore/src/validation/MSIDAuthority.m]: I see, this makes sense then. Thanks!

File: IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m [hunk 3]
```diff
@@ -140,27 +140,32 @@ - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
...
-                                    legacyUserId:(NSString *)legacyUserId
+                              legacyUserId:(NSString *)legacyUserId
                                    context:(id<MSIDRequestContext>)context
                                      error:(NSError **)error
 {
-    if (homeAccountId)
+    NSString *enrollmentId = nil;
+    
+    // If homeAccountID is provided, try to match by it first.
+    if (homeAccountId != nil)
```
Comment (oldalton) [inline IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m]: nit: no need to check for "!= nil", just if (homeAccountId) will work

File: IdentityCore/src/validation/MSIDAADAuthority.m [hunk 4]
```diff
@@ -191,6 +192,15 @@ + (instancetype)aadAuthorityWithEnvironment:(NSString *)environment
     return authority;
 }
 
+- (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
+                              legacyUserId:(NSString *)legacyUserId
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDAADAuthority.m]: nit: pass in error

File: IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m (hunk 3 above)
Comment (rokhrest) [reply IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m]: Right.
It's Authenticator habit of always being as explicit as possible :)

File: IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m [hunk 5]
```diff
@@ -140,27 +140,32 @@ - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
...
-        // If legacy userID is provided and we didn't find an exact match, do a fallback to any enrollment ID to support no userID or single userID scenarios
-        NSString *enrollmentID = legacyUserId ? [self enrollmentIdForUserId:legacyUserId context:context error:error] : nil;
-        if (enrollmentID)
-        {
-            return enrollmentID;
-        }
-        
-        enrollmentID = [self enrollmentIdIfAvailableWithContext:context error:error];
-        return enrollmentID;
+        return enrollmentId;
```
Comment (unpluggedk) [inline IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m]: nit: wouldn't this if check be better handled in the above { }?

File: IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m [hunk 6]
```diff
@@ -140,27 +140,32 @@ - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
...
     }
+    
+    // If legacy userID is provided, try to match by userID.
+    enrollmentId = legacyUserId ? [self enrollmentIdForUserId:legacyUserId context:context error:error] : nil;
+    if (enrollmentId)
+    {
+        return enrollmentId;
+    }
+
```
Comment (unpluggedk) [inline IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m]: nit: we are checking too many things above.
why not just do if (legacyUserId) { enrollmentId = ... ;  return this if not nil}
followed by returning [self enrollmendIdAva...]

Comment (rokhrest) [reply IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m]: You're right :) I just edited the code we had in this place before.

**Pull request merged**